library/src/main/java/com/datatorrent/lib/io/jms/JMSBase.java

```diff
@@ -76,18 +76,24 @@
   private transient Session session;
   private transient Destination destination;
 
+  private ConnectionFactoryBuilder connectionFactoryBuilder;
   private String connectionFactoryClass;
   private Map<String, String> connectionFactoryProperties = Maps.newHashMap();
   private String ackMode = "CLIENT_ACKNOWLEDGE";
-  private String clientId = "TestClient";
-  private String subject = "TEST.FOO";
+  private String clientId;
+  private String subject;
```

Comment:
Subject is used only in the topic mode (pub-sub mode) and only with ActiveMQ since SQS supports only the queue mode. So in the topic mode subject has to be non-null but that part of the code hasn't changed and is not even being tested (previous JMSInputOperator only had tests for the queue mode). The only thing I did was to remove the hard-coding of subject to "TEST.FOO". I'll create a JIRA to do the constraint check in the topic mode.